### PR TITLE
Spec generator - Respect tests.toml

### PR DESCRIPTION
**Pull request by ryanplusplus**
In https://github.com/exercism/lua/pull/523, @keiravillekode mentioned that the `reimplements` was handled by manually updating the generated spec. This updates the spec generator so that only tests included via `tests.toml` are included in the generated spec.

### CODE REVIEW

File: exercises/practice/book-store/book-store_spec.lua [hunk 1]
```diff
@@ -76,12 +76,11 @@ describe('book-store', function()
...
   end)
 
-  -- LuaFormatter off
-  it('check that groups of four are created properly even when there are more groups of three than groups of five', function()
+  it('check that groups of four are created properly even when there are more groups of three than groups of five',
+     function()
     local basket = { 1, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 2, 3, 3, 3, 3, 3, 3, 4, 4, 5, 5 }
     assert.are.same(14560, book_store.total(basket))
   end)
-  -- LuaFormatter on
```
Comment (ryanplusplus) [inline exercises/practice/book-store/book-store_spec.lua:84]: I re-generated all specs to make sure that we hadn't inadvertently included tests that shouldn't have been included. When I did this it reverted the change that was made in https://github.com/exercism/lua/pull/513 to fix an issue with the test runner that has since been resolved.

File: exercises/practice/state-of-tic-tac-toe/state-of-tic-tac-toe_spec.lua [hunk 2]
```diff
@@ -65,15 +65,6 @@ describe('state-of-tic-tac-toe', function()
...
     end)
 
-    it('finished game where x won via middle row victory', function()
-      local board = {
-        'O O', --
-        'XXX', --
-        ' O ' --
-      }
-      assert.are.same('win', state_of_tic_tac_toe.gamestate(board))
-    end)
```
Comment (ryanplusplus) [inline exercises/practice/state-of-tic-tac-toe/state-of-tic-tac-toe_spec.lua:75]: I think this is the test that was manually removed. It must have slipped back in the next time the spec generator was run. We should be protected against this now.

**Comment by ryanplusplus** [comment]
> This PR touches files which potentially affect the outcome of the tests of an exercise. This will cause all students' solutions to affected exercises to be re-tested.
> 
> If this PR does **not** affect the result of the test (or, for example, adds an edge case that is not worth rerunning all tests for), **please add the following to the merge-commit message** which will stops student's tests from re-running. Please copy-paste to avoid typos.
> 
> ```
> [no important files changed]
> ```
> 
> For more information, refer to the [documentation](https://exercism.org/docs/building/tracks#h-avoiding-triggering-unnecessary-test-runs). If you are unsure whether to add the message or not, please ping `@exercism/maintainers-admin` in a comment. Thank you!

We can safely merge with `[no important files changed]`

File: exercises/practice/state-of-tic-tac-toe/state-of-tic-tac-toe_spec.lua (hunk 2 above)
Comment (keiravillekode) [reply exercises/practice/state-of-tic-tac-toe/state-of-tic-tac-toe_spec.lua:75]: That was a different test case I accidentally neglected to remove.

The case I removed: 
```
description = "Invalid boards -> Invalid board"
include = false
```

was essentially the same as
```
description = "Invalid boards -> Invalid board: X won and O kept playing"
```

but with a different error string. (I stopped checking the error strings, anticipating the test case might get regenerated.)


In general, does this track have a preference for or against checking error strings?

**Review by keiravillekode — APPROVED** [review]

File: exercises/practice/state-of-tic-tac-toe/state-of-tic-tac-toe_spec.lua (hunk 2 above)
Comment (ryanplusplus) [reply exercises/practice/state-of-tic-tac-toe/state-of-tic-tac-toe_spec.lua:75]: > In general, does this track have a preference for or against checking error strings?

At least for recent tests I've been checking the strings. I don't feel strongly one way or the other.
